### PR TITLE
Dunify project

### PR DESCRIPTION
**Pull request by psafont**
This was adapted from Marcello's and Tina's work on the vendored version in xen-api.
I have left out on purpose https://github.com/xapi-project/xen-api/commit/3f71785d6ca48474880ffcbadcd5fe2343574ba3 since this was not upstreamed at the time, so it seems it was not needed for the actual fix.

### CODE REVIEW

File: examples/dune [hunk 1]
```diff
@@ -0,0 +1,5 @@
+(executable
+  (name lspci)
+  (flags (:standard -safe-string))
```
Comment (lindig) [inline examples/dune]: Safe strings are standard in OCaml 4.08 at least - so not sure we need this explicitly.

File: pci.opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+synopsis: "Ctypes bindings to libpci for OCaml"
+maintainer: "simon.beaumont@citrix.com"
```
Comment (lindig) [inline pci.opam]: This address is no longer working. Maybe this repo should use `"xen-api@lists.xen.org"`

File: pci.opam [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+authors: "Si Beaumont"
+homepage: "https://github.com/simonjbeaumont/ocaml-pci"
```
Comment (lindig) [inline pci.opam]: Point to this repo?

File: stubgen/dune [hunk 4]
```diff
@@ -0,0 +1,23 @@
+(executables
+  (names ffi_stubgen ffi_types_stubgen)
+  (flags (:standard -safe-string))
```
Comment (lindig) [inline stubgen/dune]: Not needed any more?

File: .travis.yml [hunk 5]
```diff
@@ -1,15 +1,32 @@
...
+dist: xenial
 install:
-  - wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/master/.travis-opam.sh
-  - wget https://raw.githubusercontent.com/simonjbeaumont/ocaml-travisci-skeleton/docgen/.travis-docgen.sh
-  - wget https://raw.githubusercontent.com/simonjbeaumont/ocaml-travis-coveralls/master/travis-coveralls.sh
-script: bash -ex .travis-opam.sh && bash -ex .travis-docgen.sh && bash -ex travis-coveralls.sh
+  - wget https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/.travis-opam.sh
+script: bash -ex .travis-opam.sh
+jobs:
+  include:
```
Comment (lindig) [inline .travis.yml:8]: Is this the list of builds run on Travis? It seems a bit excessive. Maybe limit this to 4.07 to 4.10?

File: bindings/dune [hunk 6]
```diff
@@ -0,0 +1,6 @@
+(library
+  (name pci_bindings)
+  (public_name pci.bindings)
+  (flags (:standard -safe-string))
```
Comment (lindig) [inline bindings/dune]: Not needed for recent OCaml versions

File: .travis.yml (hunk 5 above)
Comment (psafont) [reply .travis.yml:8]: All Ocaml versions are supported upstream, starting from 4.02. There is even one fix you backported so the code would work in versions prior to 4.05.

**Review by lindig — APPROVED** [review]

**Comment by lindig** [comment]
Are these left to do after the merge?
* Switch xapi to use this and remove the vendored package?
* Add this to xs-opam?

**Comment by psafont** [comment]
Yep, those are to do.

I checked if the xapi spec file installed any binary, but this isn't happening. It's only used as a library as far as I can see.

**Comment by lindig** [comment]
I'd be ready to merge this but maybe have a second pair of eyes take a look.

**Review by lippirk — APPROVED** [review]

**Comment by psafont** [comment]
I'll tag these changes as `v1.0.2` so it can be added to `xs-opam`.

**Pull request merged**